pkg/operator/webhook/validation/extension/handler.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package extension
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	operatorv1alpha1 "github.com/gardener/gardener/pkg/apis/operator/v1alpha1"
+)
+
+// Handler performs validation.
+type Handler struct{}
+
+func validateDelete(extension *operatorv1alpha1.Extension) (admission.Warnings, error) {
+	for _, conditionType := range []gardencorev1beta1.ConditionType{
+		operatorv1alpha1.ExtensionRequiredRuntime,
+		operatorv1alpha1.ExtensionRequiredVirtual,
+	} {
+		requiredCondition := v1beta1helper.GetCondition(extension.Status.Conditions, conditionType)
+		if requiredCondition != nil && requiredCondition.Status == gardencorev1beta1.ConditionTrue {
+			return nil, fmt.Errorf("extension is still being required: %q", requiredCondition.Message)
+		}
+	}
+
+	return nil, nil
+}
+
+// ValidateDelete performs the validation.
+func (h *Handler) ValidateDelete(_ context.Context, obj runtime.Object) (admission.Warnings, error) {
+	extension, ok := obj.(*operatorv1alpha1.Extension)
+	if !ok {
+		return nil, fmt.Errorf("expected *operatorv1alpha1.Extension but got %T", obj)
+	}
+
+	return validateDelete(extension)
```

Comment:
Does this `validateDelete` function have any purpose/benefit? Couldn't we inline its logic here?

Reply:
I made it consistent to other handlers, e.g. `pkg/operator/webhook/validation/garden/handler.go`. Apart from that, it doesn't have a special meaning.

Reply:
Can you correct it there as well? I don't think the extra functions improve readability or anything. Since they are only called once, inlining the logic seems reasonable.

Reply:
Agree, I moved the logic.